core/src/main/java/com/sequenceiq/cloudbreak/authorization/PermissionCheckerService.java

```diff
@@ -92,7 +96,7 @@ private Object checkPermission(ProceedingJoinPoint proceedingJoinPoint, MethodSi
         }
 
         try {
-            User user = userService.getCurrentUser();
+            User user = userService.getOrCreate(restRequestThreadLocalService.getIdentityUser());
```

Comment:
Is it better because the user of ```userService # getOrCreate``` is going to be aware that it only works if we have security context, e.g. the call chain is originated from the API. Are there other reasons?